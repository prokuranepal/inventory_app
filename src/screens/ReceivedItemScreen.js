import React, { useEffect, useState } from 'react';
import {
    View,
    FlatList,
    Text,
    Platform,
    ActivityIndicator,
    StyleSheet,
    Button
} from 'react-native';
import { useSelector, useDispatch } from 'react-redux';
import { HeaderButtons, Item } from 'react-navigation-header-buttons';
import moment from 'moment';
import HeaderButton from '../components/Component/HeaderButton';
import OrderItem from '../components/UI/OrderItem';
import * as itemsActions from '../store/actions/items';
import Colors from '../constants/Colors';
export const date = new Date();
export const idRan= Math.random();
const ReceivedItemScreen = props => {
    // const [isLoading, setIsLoading] = useState(false);
    const data = props.navigation.getParam('data');
    receiveditems = [
        Order = {
            date: moment(date).format('MMMM Do YYYY, hh:mm'),
            // id: "i" + Math.random(),
            items: [
                {
                    itemid: "i" + idRan,
                    itemPrice: 20,
                    quantity: 5,
                    itemTitle: data
                },
            ]
        }]


    // console.log(receiveditems)
    const dispatch = useDispatch();
    const sendAddToInventoryHandler = () => {
        dispatch(
            itemsActions.addItems(
                data,
                'BGuys',
                5,
                'Paracetamol (acetaminophen) is a pain reliever and a fever reducer',
                30,
                'General'

            )
        );
        alert(`Items has been added to inventory`);
        props.navigation.goBack();
    }


    const sendCancelHandler = () => {
        props.navigation.navigate('Scanner')
    };

    if (receiveditems.length === 0) {
        return (
            <View data-test="receivedComp" style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
                <Text>No received item  found.</Text>
            </View>
        );
    }

    return (
        <View style={styles.centered}>
            <Text style={styles.title}> Order Details</Text>
            <FlatList
                data={receiveditems}
                keyExtractor={(item => item.id)}
                data-test="flatListComp"
                renderItem={itemData => (
                    <OrderItem
                    data-test="orderItem"
                        // amount={itemData.item.totalAmount}
                        date={itemData.item.date}
                        items={itemData.item.items}
                    />
                )}
            />
            <View style={styles.buttonContainer}>
                {/* {isLoading ? (
                    <ActivityIndicator size="small" color={Colors.primary} />
                ) : ( */}
                <Button
                data-test="buttonComp"
                    color={Colors.accent}
                    title="ADD TO INVENTORY"
                    onPress={sendAddToInventoryHandler} />
                {/* )} */}
            </View>
            <View style={styles.buttonContainer}>
                <Button title="CANCEL" 
                data-test="buttonComp2"
                onPress={sendCancelHandler} />
            </View>
        </View>
    );
};

ReceivedItemScreen.navigationOptions = navData => {
    return {
        headerTitle: 'Your Received Orders',
        headerStyle: {
            backgroundColor: Platform.OS === 'android' ? Colors.primaryColor : ''
        },
        headerTitleStyle: {
            fontFamily: 'open-sans'
        },
        // headerLeft: () => (
        //     <HeaderButtons HeaderButtonComponent={HeaderButton}>
        //         <Item
        //             title="Menu"
        //             iconName={Platform.OS === 'android' ? 'md-menu' : 'ios-menu'}
        //             onPress={() => {
        //                 navData.navigation.toggleDrawer();
        //             }}
        //         />
        //     </HeaderButtons>
        // )
    };
};

const styles = StyleSheet.create({
    centered: {
        flex: 1,
        width: '100%'
    },
    buttonContainer: {
        justifyContent: 'center',
        alignItems: 'center',
        marginBottom: 20
    },
    detailItems: {
        width: '100%'
    },
    title: {
        fontSize: 20,
    },

});

export default ReceivedItemScreen;
