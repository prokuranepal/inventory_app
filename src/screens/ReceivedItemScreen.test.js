import React from 'react';
import ReceivedItemScreen ,{date,idRan} from './ReceivedItemScreen';
import {
    configure,
    mount,
} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import  {findByTestAttr, mockPlatform, findByTestProps, mockAlert} from '../_test_/componenets/test_function';
// import Alert from 'Alert';
import OrderItem from '../components/UI/OrderItem';
import moment from 'moment';

import { Provider } from 'react-redux';
// import {act} from 'react-dom/test-utils';
// JestHook.mock('expo-font');
import configureStore from 'redux-mock-store';
import TestRenderer from 'react-test-renderer';
const {act} = TestRenderer;
const mockStore = configureStore([]);

configure({
    adapter: new EnzymeAdapter
})
const function_click=jest.fn()
const dummy ={
    item:{
    date:"20 Nov, 2020",
      items:[{
        id:'id3',
    itemTitle:'Antibiotics',
    quantity:20,
    type:"general",
    itemPrice:45
      }]
    }}
  const d_func= (data1)=>{
        switch(data1){
            case('data'):
            return "data1234"
            default:
                return 
    }}
const dummy_data = {
    navigation:{
        getParam:d_func,
        setParams:function_click, 
        navigate:function_click,toggleDrawer:function_click,
        goBack:function_click
    }

}


describe('<ReceivedItemScreen />', () => {
     let store;
 
    beforeEach(() => {
      store = mockStore({
   
      });
      store.dispatch=jest.fn();
        mockPlatform('android')  
        mockAlert();
    });
 
    afterEach(() => {
        jest.clearAllMocks();
      });
         it("snapshot test in ReceivedItemScreen", () =>
        {
            const wrapper = mount(<Provider store={store}><ReceivedItemScreen {...dummy_data} /></Provider> )
            expect(wrapper).toMatchSnapshot()       
         });

         it("Props test", () =>
        {
            const wrapper = mount(<Provider store={store}><ReceivedItemScreen {...dummy_data}  /></Provider> )
            let flatListComp = findByTestAttr(wrapper,"flatListComp").at(0)
            expect(flatListComp.prop('data')).toEqual([{"date": moment(date).format('MMMM Do YYYY, hh:mm')
            , "items": [{"itemPrice": 20, "itemTitle": "data1234", "itemid": "i"+idRan, "quantity": 
            5}]}])
        });
           it("renderitem in ReceivedItemScreen", () =>
        {
            const wrapper = mount(<Provider store={store}><ReceivedItemScreen  {...dummy_data} /></Provider> )
            let flatListComp = findByTestAttr(wrapper,"flatListComp").at(0)
            let renderItem = flatListComp.props().renderItem(dummy)
            expect(renderItem.type).toEqual(OrderItem)
            expect(renderItem.props.date).toEqual("20 Nov, 2020")
            expect(renderItem.props.items).toEqual([{
              id:'id3',
          itemTitle:'Antibiotics',
          quantity:20,
          type:"general",
          itemPrice:45
            }])        });

            it("events button2 in ReceivedItemScreen", () =>
            {
                const wrapper = mount(<Provider store={store}><ReceivedItemScreen  {...dummy_data} /></Provider> )
                let buttonComp2 = findByTestAttr(wrapper, "buttonComp2");
                buttonComp2.props().onPress()
                expect(function_click).toHaveBeenCalledTimes(1)
                expect(function_click).toHaveBeenCalledWith('Scanner')
            });
            it("events button1 in ReceivedItemScreen", () =>
            {
                const wrapper = mount(<Provider store={store}><ReceivedItemScreen  {...dummy_data} /></Provider> )
                let buttonComp = findByTestAttr(wrapper, "buttonComp");
                buttonComp.props().onPress()
                expect(store.dispatch).toHaveBeenCalledTimes(1)
                // expect(store.dispatch).toHaveBeenCalledWith({'iid':moment(date).format('MMMM Do YYYY, hh:mm'), "itemData": {"company": "BGuys", "description": "Paracetamol (acetaminophen) is a pain reliever and a fever reducer", "price": 30, "quantity": 5, "title": "data1234", "type": "General"}, "type": "ADD_ITEMS"})
                expect(function_click).toHaveBeenCalledTimes(1)
            });
        });


