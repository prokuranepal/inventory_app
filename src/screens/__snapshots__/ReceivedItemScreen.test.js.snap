// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReceivedItemScreen /> snapshot test in ReceivedItemScreen 1`] = `ReactWrapper {}`;
